cookbook/product/save-product.rst

```diff
@@ -8,16 +8,48 @@ The product saver is a service, you can fetch it from the container.
 
 .. code-block:: php
 
-    $updater = $this->getContainer()->get('pim_catalog.saver.product');
+    $saver = $this->getContainer()->get('pim_catalog.saver.product');
 
 Save the products
 -----------------
 
+It implements SaverInterface and BulkSaverInterface so you can save one or many products.
+
+.. code-block:: php
+
+    $saver->save($product); // To save one product
+    $saver->saveAll($products); // To save a collection of products
+
+You can use following extra options as second parameter when you save products.
+
+.. code-block:: php
+
+    //'flush'       => be able to persist (Doctrine meaning) without flush
+    //'recalculate' => compute the completeness for the product (more greedy)
+    //'schedule'    => schedule the compute of the completeness for the product (more efficient)
+    $saver->save($product, ['flush' => true, 'recalculate' => true, 'schedule' => true]);
+
+.. warning::
+
+  Internaly, the Saver uses a persist() and flush() from Doctrine ObjectManager.
+
+  Since the 1.4 we use the changeTrackingPolicy: DEFERRED_EXPLICIT in the mapping of almost every objects, means that you need to do an explicit persist to see an object updated in database during a flush (no more dirty checking in the unit of work).
```

Comment:
we should provide an example

```
before

$object = $repository->find(12);
$object->setFoo('bar');
$em->flush();
// the property foo of the object has been set to bar without telling the entity manager to persist the product

now

$object = $repository->find(12);
$object->setFoo('bar');
// now we explicitely need to tell Doctrine to persist the object so that the changes are saved into database
$em->persist($object);
$em->flush();